src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesDeclarativeAgentUnitTest.java

```diff
@@ -0,0 +1,90 @@
+package org.csanchez.jenkins.plugins.kubernetes.pipeline;
+
+import org.csanchez.jenkins.plugins.kubernetes.pod.retention.Never;
+import org.csanchez.jenkins.plugins.kubernetes.pod.yaml.Merge;
+import org.csanchez.jenkins.plugins.kubernetes.volumes.workspace.DynamicPVCWorkspaceVolume;
+import org.jenkinsci.plugins.pipeline.modeldefinition.generator.AgentDirective;
+import org.jenkinsci.plugins.pipeline.modeldefinition.generator.DirectiveGeneratorTester;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.isA;
+
+public class KubernetesDeclarativeAgentUnitTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
```

Comment:
Not really a unit test if using `JenkinsRule`.

Would `@ClassRule` be appropriate?

Reply:
Yes, this is by opposition to `KubernetesDeclarativeAgentTest` which is performing builds, so need an actual Kubernetes environment. Can probably use `@ClassRule` here.